### DISCUSSION TITLE
Is it possible to have custom names in font sizes config ?

### DISCUSSION THREAD

**Discussion by lorcan-codes** [tailwindlabs/tailwindcss#12983] [Help] [answered]
Hi, is it possible to provide custom names in font size config in addition to existing xl, 2xl etc?

For example, I have fontSize config below. But using `text-h1` does nothing at all.

```js
   fontSize: {
        h1: "48px",
        h2: "30px",
        h3: "24px",
   }
```

If it is not possible, is there any workaround?
I assume it might be related to [text keyword ambiguities](https://tailwindcss.com/docs/adding-custom-styles#resolving-ambiguities).

**Comment by wongjn** [comment, marked as answer]
Yes, it is possible to provide custom names in the `fontSize` configuration. Your configuration is correct and should work, as evidenced by [this Tailwind Play](https://play.tailwindcss.com/bCNworf5KK?file=config). This would imply there is some other misconfiguration/misuse in your project. Consider checking:
- Tailwind has compiled since you customized the configuration.
- The `content` file globs sufficiently cover the file that the code appears in.
- [String concatenation/interpolation is not used to build the class name](https://tailwindcss.com/docs/content-configuration#dynamic-class-names). 
- `fontSize` object is declared at `theme.fontSize` or `theme.extends.fontSize` in your Tailwind configuration.
- You do not have `corePlugins.fontSize` set to `false`.
- You do not have `corePlugins` set to an array that does not include `fontSize`.

**Reply by lorcan-codes** [reply]
Hi thanks for your answer.
I went through your check list and everything checks out.

- tailwind has compiled after my changes because I see color changes immediately 
- Content covers my components. Otherwise I would not see color changes
- String concatenation/interpolation is not used. I am using `text-h1`
- I am declaring it at `theme.extends.fontSize` but tried with `theme.fontSize` as well. Same results
- `corePlugins.fontSize` is not set to false unless it is false by default
- I have corePlugins set to object with preflight set to false

```js
  corePlugins: {
    preflight: false,
  },
```

I will debug my code further and try with a new brand new repo if I can't get it to work.

**Reply by lorcan-codes** [reply]
I created a tailwind project and it worked fine.  
I was using shadcn so I created a new project with shadcn to test it. It seems the issue with shadcn. 
They are using `twMerge` method from `tailwind-merge` in their `cn` utility function. 

Here is my shadcn repo with a bug if you want to take a look at.

GH: https://github.com/lorcan-codes/vite-shadcn
Preview: https://vite-shadcn.vercel.app/

**Reply by wongjn** [reply]
You'd probably want different font sizes as a variant group within the `cva()` call in the `Button` component.

**Reply by lorcan-codes** [reply]
I found a way how to make it work. 
https://github.com/lorcan-codes/vite-shadcn/commit/af3ca21072d106b036304388de659818fc35520d
